### ISSUE TITLE
Remove 300ms delay

### ISSUE THREAD

**Issue by jelmervdl** [XapaJIaMnu/translateLocally#41]
@kpu said:
> The current setup is killing latency when we have a 300 ms wait then a 20 ms translation time and brag about how fast that 20 ms was.

Why display the speed if there is always a noticeable 300ms delay.

The 300ms delay is there to prevent unnecessary translations from being triggered, the idea being that 300ms is about the time the user needs to switch from typing to reading the translation. By holding off the translation, it can be started immediately once the user does stop typing instead of having to wait for a translation in progress before starting to translate that final input.

## Implementation
This only really makes sense if the translation takes longer than 300ms itself. So I suggest we set the initial delay to 1ms, and once the input text is large enough that translation time is longer than 300ms, we switch input delay to 300ms. That way you get both the noticeable speed initially, but also quick response times when there is a lot of text (or a really slow machine).

Things to take into account:
- measure over a couple of translations (the first translation is always a bit slower than the following ones)
- prevent weird inconsistent switching behaviour when right on that 300ms edge
- reset the delay to 1ms when text field is cleared?
- Can we base it on the words/second speed measurement? That way we can estimate whether the delay is applicable before changing it (would solve the text field cleared -> go back to 1ms delay case automatically)

## Scope
This can further be improved a lot by only translating sentences we haven't already translated, but that's out of scope for this issue.

**Comment by XapaJIaMnu** [comment]
I think for now, we can change it to 1 ms (and probably add a setting in the configuration). My reasoning is, that people that copy/paste long sequences of text wouldn't edit them inside the app, but somewhere else.

**Comment by jelmervdl** [comment]
I'll remove the timer entirely for the time being. There isn't much added value of having a signal trigger a timer trigger a signal, all on the same thread anyway, and removing it makes the code a little simpler. Any improvement in responsiveness further down the line will probably come in the form of translating only sentences that changed.
